src/internal/storage/chunk/reader.go

```diff
@@ -60,9 +66,36 @@ func (r *Reader) Iterate(cb func(*DataReader) error) error {
 }
 
 // Get writes the concatenation of the data referenced by the data references.
-func (r *Reader) Get(w io.Writer) error {
+func (r *Reader) Get(w io.Writer) (retErr error) {
+	if len(r.dataRefs) <= 1 {
+		return r.Iterate(func(dr *DataReader) error {
+			return dr.Get(w)
+		})
+	}
+	ctx, cancel := context.WithCancel(r.ctx)
```

Comment:
You should do a `defer cancel()` here.  I think it leaks below when retErr == nil.  It's safe to call more than once.